client/src/templates/Challenges/utils/frame.ts

```diff
@@ -267,9 +267,44 @@ const writeContentToFrame = (frameContext: Context) => {
     createHeader(frameContext.element.id) + frameContext.build,
     frameContext.document
   );
+
+  registerScrollEventListner(frameContext.element);
+
+  if (getPreviewScrollPosition()) {
+    restorePreviewScrollPosition(
+      frameContext.element,
+      getPreviewScrollPosition()
+    );
+  }
   return frameContext;
 };
 
+let previewScrollPosition = 0;
+export const getPreviewScrollPosition = () => previewScrollPosition;
+export const setPreviewScrollPosition = (position: number) =>
+  (previewScrollPosition = position);
+
+const registerScrollEventListner = (iframe: HTMLIFrameElement) => {
+  iframe.contentDocument?.addEventListener(
+    'scroll',
+    (event: IframeEvent<Document>) => {
+      const { target } = event;
+      if (target.body.scrollTop) {
+        setPreviewScrollPosition(target.body.scrollTop);
+      }
+    }
+  );
+};
+
+const restorePreviewScrollPosition = (
+  iframe: HTMLIFrameElement,
+  position: number
+) => {
+  if (iframe.contentDocument?.body) {
+    iframe.contentDocument.body.scrollTop = position;
+  }
+};
+
```

Comment:
Rather than have these be globals, I think this is a good time to use a class.  That class, maybe `ScrollManager`, can hold `previewScrollPosition` and have `registerScrollEventListner` and `restorePreviewScrollPosition` as methods.

We'd still be polluting the global scope, but less so. Also, it's very clear that what all these various functions and variables are for.

Reply:
@ojeytonwilliams Where should I keep that class? In this file or make a separate one?

Reply:
I added the class in the same file for now.

Reply:
It's only used here, so that works.